elixir/apps/web/lib/web/live/sites/new_token.ex

```diff
@@ -295,6 +220,81 @@ defmodule Web.Sites.NewToken do
+    if [ ! -e /usr/local/bin/firezone-gateway ]; then
+      FIREZONE_VERSION=\\$(curl -Ls \\\\
+        -H "Accept: application/vnd.github+json" \\\\
+        -H "X-GitHub-Api-Version: 2022-11-28" \\\\
+        "https://api.github.com/repos/firezone/firezone/releases/latest" | grep '"tag_name":' | sed 's/.*"tag_name": "\([^"]*\).*/\1/'
+      )
+      [ "\\$FIREZONE_VERSION" = "" ] && echo "[Error] Cannot fetch latest version. Rate-limited by GitHub?" && exit 1
+      echo "Downloading Firezone Gateway version \\$FIREZONE_VERSION"
+      arch=\\$(uname -m)
+      case \\$arch in
+        aarch64)
+          bin_url="https://github.com/firezone/firezone/releases/download/\\$FIREZONE_VERSION/gateway-arm64"
+          ;;
+        armv7l)
+          bin_url="https://github.com/firezone/firezone/releases/download/\\$FIREZONE_VERSION/gateway-arm"
+          ;;
+        x86_64)
+          bin_url="https://github.com/firezone/firezone/releases/download/\\$FIREZONE_VERSION/gateway-x64"
+          ;;
+        *)
+          echo "Unsupported architecture"
+          exit 1
+      esac
+      curl -Ls \\$bin_url -o /usr/local/bin/firezone-gateway
+    fi
+
+    # Set proper capabilities and permissions on each start
+    chgrp firezone /usr/local/bin/firezone-gateway
+    chmod 0750 /usr/local/bin/firezone-gateway
+    setcap 'cap_net_admin+eip' /usr/local/bin/firezone-gateway
+    mkdir -p /var/lib/firezone
+    chown firezone:firezone /var/lib/firezone
+    chmod 0775 /var/lib/firezone
+
+    # Enable masquerading for ethernet and wireless interfaces
+    iptables -C FORWARD -i tun-firezone -j ACCEPT 2&>1 > /dev/null || iptables -A FORWARD -i tun-firezone -j ACCEPT
+    iptables -C FORWARD -o tun-firezone -j ACCEPT 2&>1 > /dev/null || iptables -A FORWARD -o tun-firezone -j ACCEPT
+    iptables -t nat -C POSTROUTING -o e+ -j MASQUERADE 2&>1 > /dev/null || iptables -t nat -A POSTROUTING -o e+ -j MASQUERADE
+    iptables -t nat -C POSTROUTING -o w+ -j MASQUERADE 2&>1 > /dev/null || iptables -t nat -A POSTROUTING -o w+ -j MASQUERADE
+    ip6tables -C FORWARD -i tun-firezone -j ACCEPT 2&>1 > /dev/null || ip6tables -A FORWARD -i tun-firezone -j ACCEPT
+    ip6tables -C FORWARD -o tun-firezone -j ACCEPT 2&>1 > /dev/null || ip6tables -A FORWARD -o tun-firezone -j ACCEPT
+    ip6tables -t nat -C POSTROUTING -o e+ -j MASQUERADE 2&>1 > /dev/null || ip6tables -t nat -A POSTROUTING -o e+ -j MASQUERADE
+    ip6tables -t nat -C POSTROUTING -o w+ -j MASQUERADE 2&>1 > /dev/null || ip6tables -t nat -A POSTROUTING -o w+ -j MASQUERADE
+
+    # Enable packet forwarding
+    sysctl -w net.ipv4.ip_forward=1
+    sysctl -w net.ipv4.conf.all.src_valid_mark=1
+    sysctl -w net.ipv6.conf.all.disable_ipv6=0
+    sysctl -w net.ipv6.conf.all.forwarding=1
+    sysctl -w net.ipv6.conf.default.forwarding=1
+    EOF
+
+    # Make ExecStartPre script executable
+    chmod +x /usr/local/bin/firezone-gateway-execstartpre.sh
```

Comment:
Is this a standard location for unit file scripts?

Reply:
The thinking was that this is more of an upgrade and fix-permissions script. I think `/usr/local/bin` is appropriate for those, but I think this should probably be renamed to `/usr/local/bin/firezone-gateway-init.sh` to be usable outside of the `systemd` context.

Reply:
Fixed, good catch